CMakeLists.txt

```diff
@@ -87,6 +87,9 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
                         -Wextra
                         -Wno-unused-parameter
                         -Wno-missing-field-initializers
+                        -Werror=int-conversion
+                        -Werror=pointer-to-int-cast
+                        -Werror=int-to-pointer-cast
```

Comment:
If you're doing all of the other changes, do you still need those new flags?

Reply:
The point is to avoid these happening again in the future, which will be checked on each PR when a CI is set up

Reply:
Oh apologies, I read it with `-Wno-error-*`